osgar/drivers/realsense.py

```diff
@@ -85,17 +89,25 @@ def pose_callback(self, frame):
     def depth_callback(self, frameset):
         try:
             assert frameset.is_frameset()
-            frame = frameset.as_frameset().get_depth_frame()
-            n = frame.get_frame_number()
+            depth_frame = frameset.as_frameset().get_depth_frame()
+            color_frame = frameset.as_frameset().get_color_frame()
+            n = depth_frame.get_frame_number()
             if n % self.depth_subsample != 0:
                 return
-            assert frame.is_depth_frame()
-            depth_image = np.asanyarray(frame.as_depth_frame().get_data())
+            assert depth_frame.is_depth_frame()
+            assert color_frame.is_video_frame() == self.depth_rgb
+            depth_image = np.asanyarray(depth_frame.as_depth_frame().get_data())
+            self.publish('depth', depth_image)
+
+            if color_frame.is_video_frame():
+                color_image = np.asanyarray(color_frame.as_frame().get_data())
+                __, data = cv2.imencode('*.jpeg', color_image)
+                self.publish('color', data.tobytes())
         except Exception as e:
             print(e)
             self.finished.set()
             return
-        self.publish('depth', depth_image)
```

Comment:
The `publish` call has been intentional kept outside of the overly generic try/except block that is supposed to catch any possible exception raised by pyrealsense library. The `publish` call should raise no exceptions so when it does raise something, we want it to propagate up and not swallow it.